source/NVDAObjects/behaviors.py

```diff
@@ -475,6 +476,13 @@ def _get_caretMovementDetectionUsesEvents(self):
 		prompt to be read when quickly deleting text."""
 		return False
 
+	def event_textChange(self) -> None:
+		"""Fired when the text changes.
+		@note: Updates also braille.
+		"""
+		super().event_textChange()
+		braille.handler.handleUpdate(self)

```

Comment:
Note the `@note: It is safe to call this directly from threads other than the main thread.` on LiveText. I think this might be problematic for the legacy winconsole implementation as another thread will try to update the braille handler.

Reply:
There is a lock in use. After adding it as far as I know I have not encountered race condition.

Reply:
I don't think a lock is a good idea here. If a method (event-texctChange) requires thread safety, the implementation should always be thread safe. I would personally remove the lock and go for an alternative for the time being, see my next comment.